### PR TITLE
dm: support MySQL 8.0 GA

### LINKED ISSUE #10405: DM supports MySQL 8.0 GA
## Enhancement
Currently descring of DM as below:
Database version requirements
MySQL version 5.5 ~ 5.7
MySQL version 8.0 (experimental features)
As EOL of Aurora2 (MySQL5.7) is approaching, customer is considering if there is any way to migrate from Aurora3 (MySQL8.0) to TiDB for the future.

### PR DESCRIPTION
**Pull request by lyzx2001**
### What problem does this PR solve?

Issue Number: close #10405 

### What is changed and how it works?
DM supports MySQL 8.0 GA.

#### Tests 

 - Unit test

##### Will it cause performance regression or break compatibility?
No.
##### Do you need to update user documentation, design documentation or monitoring documentation?
https://github.com/pingcap/docs/pull/15959
https://github.com/pingcap/docs-cn/pull/15985
### Release note 

```release-note
DM supports MySQL 8.0 GA.
```

### CODE REVIEW

File: dm/pkg/checker/mysql_server.go [hunk 1]
```diff
@@ -35,14 +35,14 @@ func NewMySQLVersionChecker(db *sql.DB, dbinfo *dbutil.DBConfig) RealChecker {
...
-// * 5.6.0 <= MySQL Version < 8.0.0.
+// * 5.6.0 <= MySQL Version < 8.0.34.
 var SupportedVersion = map[string]struct {
 	Min MySQLVersion
 	Max MySQLVersion
 }{
 	"mysql": {
 		MySQLVersion{5, 6, 0},
-		MySQLVersion{8, 0, 0},
+		MySQLVersion{8, 0, 34},
```
Comment (lance6716) [inline dm/pkg/checker/mysql_server.go:45]: why do we choose this version?

https://dev.mysql.com/doc/relnotes/mysql/8.0/en/news-8-0-35.html

Comment (lyzx2001) [reply dm/pkg/checker/mysql_server.go]: right, supported version should be < `8.0.36`

Comment (lance6716) [reply dm/pkg/checker/mysql_server.go]: can you explain the reason why we choose that version? currently latest MySQL's release version?

Comment (lyzx2001) [reply dm/pkg/checker/mysql_server.go]: yes, I initially tend to support `8.0.x`, which means we support all the versions starting with `8.0`, but in the code we need to be specific about the version, so I use the latest MySQL's release version

Comment (lance6716) [reply dm/pkg/checker/mysql_server.go]: You can use `MySQLVersion{8, 1, 0}` then

Comment (lyzx2001) [reply dm/pkg/checker/mysql_server.go]: modified

**Review by lance6716 — APPROVED** [review]

**Review by D3Hunter — APPROVED** [review]

**Pull request merged**